### PR TITLE
add the download function of commonvoice command line interface

### PR DESCRIPTION
**Pull request by manbaaaa**
When the version is less than 8.0, CommonVoice requires you to enter e-mail to download the data.
Or you can choose a version greater than 8.0.

### CODE REVIEW

**Comment by pzelasko** [comment]
I see you closed it, but I'd be happy to merge if you want to finish the PR.

**Comment by manbaaaa** [comment]
When I run this command,pre-commit run, I got an error.
[INFO] Installing environment for https://github.com/pycqa/isort.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/lishaojie/.cache/pre-commit/repo0867tdpc/py_env-python3/bin/python', '-mpip', 'install', '.')
return code: 1
stdout:
    Processing /Users/lishaojie/.cache/pre-commit/repo0867tdpc
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'error'
    
stderr:
      error: subprocess-exited-with-error
      
      × Preparing metadata (pyproject.toml) did not run successfully.
      │ exit code: 1
      ╰─> [14 lines of output]
          Traceback (most recent call last):
            File "/Users/lishaojie/.cache/pre-commit/repo0867tdpc/py_env-python3/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
              main()
            File "/Users/lishaojie/.cache/pre-commit/repo0867tdpc/py_env-python3/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
              json_out['return_val'] = hook(**hook_input['kwargs'])
            File "/Users/lishaojie/.cache/pre-commit/repo0867tdpc/py_env-python3/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 149, in prepare_metadata_for_build_wheel
              return hook(metadata_directory, config_settings)
            File "/private/var/folders/1c/b41cgr692j54pvc6psgv4bp40000gn/T/pip-build-env-pqepxcm_/overlay/lib/python3.7/site-packages/poetry/core/masonry/api.py", line 40, in prepare_metadata_for_build_wheel
              poetry = Factory().create_poetry(Path(".").resolve(), with_groups=False)
            File "/private/var/folders/1c/b41cgr692j54pvc6psgv4bp40000gn/T/pip-build-env-pqepxcm_/overlay/lib/python3.7/site-packages/poetry/core/factory.py", line 57, in create_poetry
              raise RuntimeError("The Poetry configuration is invalid:\n" + message)
          RuntimeError: The Poetry configuration is invalid:
            - [extras.pipfile_deprecated_finder.2] 'pip-shims<=0.3.4' does not match '^[a-zA-Z-_.0-9]+$'
          
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
    error: metadata-generation-failed
    
    × Encountered error while generating package metadata.
    ╰─> See above for output.
    
    note: This is an issue with the package mentioned above, not pip.
    hint: See above for details.
    
Check the log at /Users/lishaojie/.cache/pre-commit/pre-commit.log

**Comment by manbaaaa** [comment]
I have solved it by modify the isort version in the pre-commit configuration file, 5.10.1-->5.12.0，I will resubmit a pr later.

File: lhotse/recipes/commonvoice.py [hunk 1]
```diff
@@ -30,7 +30,9 @@
 )
 
 DEFAULT_COMMONVOICE_URL = "https://voice-prod-bundler-ee1969a6ce8178826482b88e843c335139bd3fb4.s3.amazonaws.com"
```
Comment (pzelasko) [inline lhotse/recipes/commonvoice.py:32]: can you remove the deprecated URL here?

File: lhotse/recipes/commonvoice.py [hunk 2]
```diff
@@ -46,7 +48,7 @@ def download_commonvoice(
     languages: Union[str, Iterable[str]] = "all",
     force_download: bool = False,
     base_url: str = DEFAULT_COMMONVOICE_URL,
-    release: str = DEFAULT_COMMONVOICE_RELEASE,
+    release: Optional[str] = "",
```
Comment (pzelasko) [inline lhotse/recipes/commonvoice.py:51]: Can we go back to having a valid default release to be used?

File: lhotse/recipes/commonvoice.py [hunk 3]
```diff
@@ -62,11 +64,11 @@ def download_commonvoice(
...
-    raise NotImplementedError(
-        "CommonVoice requires you to enter e-mail to download the data"
-        "-- please download it manually for now. "
-        "We are open to contributions to support downloading CV via lhotse."
-    )
+    # raise NotImplementedError(
+    #    "CommonVoice requires you to enter e-mail to download the data"
+    #    "-- please download it manually for now. "
+    #    "We are open to contributions to support downloading CV via lhotse."
+    # )
```
Comment (pzelasko) [inline lhotse/recipes/commonvoice.py:71]: Please remove the commented out code.

**Review by pzelasko — COMMENTED** [review]
Thanks, I left a few more comments to clean up the code before merging.

**Comment by pzelasko** [comment]
Thanks!

**Pull request merged**